packages/react-core/src/components/BackToTop/__tests__/BackToTop.test.tsx

```diff
@@ -1,22 +1,93 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import { fireEvent, render, screen } from '@testing-library/react';
 import { BackToTop } from '../BackToTop';
+import userEvent from '@testing-library/user-event';
 
-describe('BackToTop', () => {
-  test('verify basic', () => {
-    const { asFragment } = render(<BackToTop />);
-    expect(asFragment()).toMatchSnapshot();
-  });
+test('Renders with the default class', () => {
+  render(
+    <div data-testid="backtotop">
+      <BackToTop />
+    </div>
+  );
 
-  test('verify custom class', () => {
-    const { asFragment } = render(<BackToTop className="custom-css">test</BackToTop>);
+  expect(screen.getByTestId('backtotop').firstChild).toHaveClass('pf-c-back-to-top');
+});
+
+test('BackToTop is not yet visible', () => {
+  render(
+    <div data-testid="backtotop">
+      <BackToTop />
+    </div>
+  );
+
+  expect(screen.getByTestId('backtotop').firstChild).toHaveClass('pf-m-hidden');
+});
+
+test('BackToTop is visible after scrolling', () => {
+  render(
+    <div data-testid="backtotop">
+      <BackToTop />
+    </div>
+  );
 
-    expect(asFragment()).toMatchSnapshot();
-  });
+  fireEvent.scroll(window, { target: { scrollY: 401 } });
+  expect(screen.getByTestId('backtotop').firstChild).not.toHaveClass('pf-m-hidden');
+});
+
+test('ScrollTo event is fired after clicking BackToTop', () => {
+  render(
+    <div data-testid="backtotop">
+      <BackToTop />
+    </div>
+  );
+
+  fireEvent.scroll(window, { target: { scrollY: 401 } });
+  global.scrollTo = jest.fn();
+
+  userEvent.click(screen.getByTestId('backtotop').firstChild as Element);
+  expect(global.scrollTo).toHaveBeenCalled();
+});
+
+test('Renders backToTop when isAlwaysVisible prop is set', () => {
+  render(
+    <div data-testid="backtotop">
+      <BackToTop isAlwaysVisible />
+    </div>
+  );
+
+  expect(screen.getByTestId('backtotop').firstChild).toBeVisible();
+  expect(screen.getByTestId('backtotop').firstChild).not.toHaveClass('pf-m-hidden');
```

Comment:
Since the earlier tests are just checking for the `pf-m-hidden` class, could we remove the `toBeVisible` line here?